plugins/uefi/fu-uefi-pcrs.c

```diff
@@ -103,10 +96,52 @@ fu_uefi_pcrs_setup_tpm12 (FuUefiPcrs *self, const gchar *fn_pcrs, GError **error
 }
 
 static gboolean
-fu_uefi_pcrs_setup_tpm20 (FuUefiPcrs *self, const gchar *argv0, GError **error)
+fu_uefi_pcrs_setup_tpm20 (FuUefiPcrs *self, GError **error)
 {
-	const gchar *argv[] = { argv0, NULL };
-	return fu_common_spawn_sync (argv, fu_uefi_pcrs_parse_line, self, 1500, NULL, error);
+	TSS2_RC rc;
+	ESYS_CONTEXT *ctx = NULL;
+	TPML_PCR_SELECTION pcr_selection_in = { 0, };
+	TPML_DIGEST *pcr_values = NULL;
+	g_autoptr(GString) str = NULL;
+	FuUefiPcrItem *item;
+
+	rc = Esys_Initialize (&ctx, NULL, NULL);
+	if (rc != TSS2_RC_SUCCESS) {
+		return FALSE;
+	}
+
+	/* currently we only use PCR 0 on banks SHA-1 and SHA-256, so fetch only these */
+	pcr_selection_in.count = 2;
+	pcr_selection_in.pcrSelections[0].hash = TPM2_ALG_SHA1;
+	pcr_selection_in.pcrSelections[0].sizeofSelect = 3;
+	pcr_selection_in.pcrSelections[0].pcrSelect[0] = 0b00000001;
+	pcr_selection_in.pcrSelections[1].hash = TPM2_ALG_SHA256;
+	pcr_selection_in.pcrSelections[1].sizeofSelect = 3;
+	pcr_selection_in.pcrSelections[1].pcrSelect[0] = 0b00000001;
+
+	rc = Esys_PCR_Read (ctx, ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+	                    &pcr_selection_in, NULL, NULL, &pcr_values);
+	Esys_Finalize (&ctx);
+	if (rc != TSS2_RC_SUCCESS) {
+		return FALSE;
```

Comment:
In this case `error` needs to be filled out.

Reply:
Done.